tests/unit/parser.js

```diff
@@ -239,28 +317,128 @@ tap.test('Parser', {autoend: true}, t => {
+        const outputFormat = 'XML'
+        const expectedHeader = {
+          'Content-Type': 'application/xml'
+        }
+
+        const expectedBody =
+          '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>\n<root>\n' +
+          '  <Name>Jet</Name>\n' +
+          '  <Surname>Li</Surname>\n' +
+          '  <Email>jet@openhim.org</Email>\n</root>'
+
+        parseOutgoingBody(ctx, outputFormat)
+
+        t.deepEqual(ctx.body, expectedBody)
+        t.deepEqual(ctx.header, expectedHeader)
+        t.equals(ctx.orchestrations.length, 1)
+        t.equals(ctx.orchestrations[0].name, 'Outgoing Parser')
+        t.end()
+      }
+    )
   })
 
-  t.test(
-    'should throw due to invalid input type and add error message to ctx',
-    async t => {
+  t.test('parseBodyMiddleware', {autoend: true}, t => {
+    t.test(
+      'should return an error for invalid input type and add error message to ctx',
+      async t => {
+        const ctx = {
+          state: {
+            uuid: 'unique-id',
+            metaData: {
+              name: 'Test endpoint',
+              transformation: {
+                input: 'XML',
+                output: 'JSON'
+              }
+            }
+          }
+        }
+
+        const next = () => {
+          t.equals(ctx.status, 400)
+          t.same(ctx.body, {
+            error:
+              'Test endpoint (unique-id): transformation method "INVALID" not yet supported'
+          })
+        }
+        await parseBodyMiddleware()(ctx, next)
+      }
+    )
+
+    t.test('should complete middleware request', async t => {
       const ctx = {
+        body: {
+          Name: 'Jet',
+          Surname: 'Li',
+          Email: 'jet@openhim.org'
+        },
         state: {
+          uuid: 'randomUidForRequest',
+          allData: {},
           metaData: {
+            name: 'Testing endpoint',
             transformation: {
-              input: 'invalid',
-              output: 'JSON'
+              input: 'JSON',
+              output: 'XML'
             }
           }
-        }
+        },
+        get: name => {
+          const contentHeader = 'application/json'
+          if (name) {
+            return contentHeader
+          }
+        },
+        set: (key, value) => {
+          ctx.header = {}
+          ctx.header[key] = value
+        },
+        request: {}
       }
 
-      await parseBodyMiddleware()(ctx)
-      t.equals(ctx.status, 400)
-      t.same(ctx.body, {
-        error:
-          'undefined (undefined): transformation method "INVALID" not yet supported'
-      })
-    }
-  )
+      const next = () => {}
+      await parseBodyMiddleware()(ctx, next)
+
+      t.equals(
+        ctx.body,
+        '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>\n' +
+          '<root>\n' +
+          '  <Name>Jet</Name>\n' +
+          '  <Surname>Li</Surname>\n' +
+          '  <Email>jet@openhim.org</Email>\n' +
+          '</root>'
+      )
+
+      t.pass()
```

Comment:
Is that t.pass() at the end in the right place? Would it be better to test that the next middleware is called?
And add a t.plan(2)
```suggestion
      const next = () => {
        t.pass()
      }
      await parseBodyMiddleware()(ctx, next)

      t.equals(
        ctx.body,
        '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>\n' +
          '<root>\n' +
          '  <Name>Jet</Name>\n' +
          '  <Surname>Li</Surname>\n' +
          '  <Email>jet@openhim.org</Email>\n' +
          '</root>'
      )
```

Reply:
I think the t.pass was scaffolding the test and has no value there. This can be removed as there is already an asset checking the ctx.body is correct

Reply:
done in 6f482aa